demo-app/src/main/java/cash/z/ecc/android/sdk/demoapp/ui/screen/seed/view/ConfigureSeedView.kt

```diff
@@ -113,5 +130,88 @@ private fun ConfigureSeedMainContent(
         ) {
             Text(text = stringResource(id = R.string.seed_random))
         }
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        RestoreWalletSection(zcashNetwork, onExistingWallet)
+    }
+}
+
+@Composable
+@Suppress("ktlint:standard:function-naming")
+private fun RestoreWalletSection(
+    zcashNetwork: ZcashNetwork,
+    onExistingWallet: (PersistableWallet) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Column(modifier = modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(8.dp)) {
+        val context = LocalContext.current
+
+        var seedPhraseString by rememberSaveable {
+            mutableStateOf("")
+        }
+        var walletBirthdayString by rememberSaveable {
+            mutableStateOf("")
+        }
+        var requiredDataIsEmpty by rememberSaveable {
+            mutableStateOf(true)
+        }
+
+        fun preValidateInput() {
+            requiredDataIsEmpty = seedPhraseString.isEmpty()
+        }
+
+        fun validateAndRestore() {
+            try {
+                val seedPhrase = SeedPhrase.new(seedPhraseString)
+
+                val blockHeight =
+                    walletBirthdayString.toLongOrNull()?.let { blockHeight ->
+                        BlockHeight.new(zcashNetwork, blockHeight)
+                    } ?: WalletFixture.lastKnownBirthday(zcashNetwork) // fallback to last known valid value
```

Comment:
In a real-world wallet, we should rather rely on the `zcashNetwork.saplingActivationHeight` instead of fallbacking to the known birthdays of the fixture wallets to avoid omitting wallet transactions.

Reply:
That was a good idea. But in this case, we need to stick with what the user typed in (or did not). Later, we could consider adding, for example, a new button `Restore from latest checkpoint` or so, to intentionally avoid a long time initial sync for an older wallet.